internal/datastore/spanner/readwrite.go

```diff
@@ -177,7 +186,16 @@ func deleteWithFilter(ctx context.Context, rwt *spanner.ReadWriteTransaction, fi
 
 func upsertVals(r *core.RelationTuple) []interface{} {
 	key := keyFromRelationship(r)
-	return append(key, spanner.CommitTimestamp)
+	key = append(key, spanner.CommitTimestamp)
+	if r.Caveat != nil && len(r.Caveat.CaveatName) > 0 {
+		key = append(key, r.Caveat.CaveatName)
+		if r.Caveat.Context != nil {
```

Comment:
shouldn't we appending a `nil` context if it's empty? we seem to be doing it in the else block below?